core/utilities/LogManager.cpp

```diff
@@ -0,0 +1,380 @@
+
+    logFiles.erase(std::remove_if(logFiles.begin(), logFiles.end(),
+                                  [&](const std::string& file) -> bool {
+                                      return std::chrono::duration<double>(std::difftime(
+                                               std::chrono::system_clock::to_time_t(std::chrono::system_clock::now()),
+                                               FileSystemUtils::getLastModified(file))) < m_uploadAfter;
+                                  }),
+                   logFiles.end());
+
+    return logFiles;
+}
+
+std::vector<std::string> LogManager::getLogFiles()
+{
+    std::vector<std::string> logFiles;
+
+    logFiles = FileSystemUtils::listFiles(m_logDirectory);
+    logFiles.erase(std::remove_if(logFiles.begin(), logFiles.end(),
+                                  [&](const std::string& file) -> bool {
+                                      return !wolkabout::StringUtils::endsWith(file, m_logExtension);
+                                  }),
+                   logFiles.end());
+    return logFiles;
+}
+
+const std::chrono::hours& LogManager::getUploadAfter() const
+{
+    return m_uploadAfter;
+}
+
+void LogManager::setUploadAfter(const std::chrono::hours& uploadAfter)
+{
+    m_uploadAfter = uploadAfter;
+
+    if (m_uploadAfter != std::chrono::hours(0))
+    {
+        restart();
+    }
+}
+
+void LogManager::uploadLogs()
+{
+    if (m_uploadAfter == std::chrono::hours(0))
+    {
+        LOG(DEBUG) << "Upload after is set to 0, not uploading logs";
+        return;
+    }
+
+    if (!m_logUploader)
+    {
+        LOG(WARN) << "No log uploader, not attempting upload of logs.";
+        return;
+    }
+
+    LOG(INFO) << "Start uploading log files";
+
+    std::vector<std::string> logs = getLogsToUpload();
+
+    for (auto& log : logs)
+    {
+        LOG(INFO) << "Uploading '" << log << "'";
+        if (!m_logUploader->upload(log))
+        {
+            LOG(WARN) << "Failed to upload '" << log << "'";
+        }
+    }
+
+    LOG(INFO) << "Ended uploading log files";
+}
+
+void LogManager::deleteOldLogs()
+{
+    if (m_deleteAfter == std::chrono::hours(0))
+    {
+        LOG(DEBUG) << "Delete after is set to 0, not deleting logs";
+        return;
+    }
+
+    LOG(INFO) << "Starting to delete stale log files";
+
+    std::vector<std::string> oldLogs = getLogsToDelete();
+
+    for (auto& oldLog : oldLogs)
+    {
+        LOG(INFO) << "Deleting '" << oldLog << "'";
+        if (!wolkabout::FileSystemUtils::deleteFile(oldLog))
+        {
+            LOG(WARN) << "Failed to delete log '" << oldLog << "'";
+        }
+    }
+    LOG(INFO) << "Ended deleting stale log files";
+}
+void LogManager::checkLogOverflow()
+{
+    if (m_maxSize == 0)
+    {
+        LOG(TRACE) << "Max log size set to 0, not checking for overflow";
+        return;
+    }
+
+    LOG(INFO) << "Checking for log overflow";
+
+    std::vector<std::string> logs = getLogFiles();
+
+    double logSize = 0;
```

Comment:
DRY :+1: 
Private method `getTotalLogSize` ?

because
"returns logsize and accept as parameter log." == `wolkabout::FileSystemUtils::getFileSize(log)`